tree/treeplayer/inc/ROOT/TArrowDS.hxx

```diff
@@ -0,0 +1,202 @@
+#ifndef ROOT_TARROWTDS
+#define ROOT_TARROWTDS
+
+#include <arrow/table.h>
+#include <memory>
+
+#include "ROOT/TDataFrame.hxx"
+#include "ROOT/TDataSource.hxx"
+
+namespace ROOT {
+namespace Experimental {
+namespace TDF {
+
+class TArrowDS final : public ROOT::Experimental::TDF::TDataSource {
+private:
+   // Per slot visitor of an Array.
+   class ArrayPtrVisitor : public arrow::ArrayVisitor {
```

Comment:
I fixed it to `::arrow::ArrayVisitor`. While `ArrayPtrVisitor` and `ValueGetter` are indeed implementation details, they cannot be hidden in the `.cxx` because `TArrowDS::fValueGetters` is of type `std::vector<ValueGetter>` and therefore needs to know the size. I can move some of the method implementations in the `.cxx`, if requested, but I'd rather do that in a separate PR, since there is no particular reason why that should delay merging this PR.

Reply:
No problem. Just keep it as is.